doc/design/var_desc.md

```diff
@@ -0,0 +1,90 @@
+## Background
+PaddlePaddle divides the description of neural network computation graph into two stages: compile time and runtime.
+
+The data structure to describe the compile time graph should be able to be serialized for distributed training. So we use proto message to describe the graph: OpDesc to describe computation and VarDesc to describe data.
+
+PaddlePaddle will generate these data structure according to user's description and do some optimization, such as:
+
+1. InferShape. Infer the Output size according to Input size and set them into VarDesc.
+1. memory optimise and reuse. Scan all the memory that will be used and reuse some memory that is allocated before but will not be used anymore to reduce memory.
+
+VarDesc is used to describe different kinds of Variable value, such as Tensor, scalar, and scope:
+
+## Definition of VarDesc in Proto
+
+```
+message LoDTensorDesc {
+  enum Type {
+    INT16 = 1;
+    INT32 = 2;
+    INT64 = 3;
+    FP16 = 4;
+    FP32 = 5;
+    DOUBLE = 6
```

Comment:
Unify float names, please.  Either `FP16, FP32, FP64`, or `half, float, double`. Do not mix them together.

Reply:
done